app/controllers/admin/links_controller.rb

```diff
@@ -174,22 +138,20 @@ def join_discord_redirect
   private
     def fetch_product_by_general_permalink
       @product = Link.find_by(id: params[:id])
-      return redirect_to admin_product_path(@product.unique_permalink) if @product
+      return redirect_to admin_product_path(@product.id) if @product
```

Comment:
If we navigate to `/admin/links/1` and there's a product for ID 1, we redirect to `/admin/products/1`